test/python/k8s/deploy_test

```diff
@@ -0,0 +1,192 @@
+ssl_cert=$(kubectl exec -c "conjur-oss-nginx" $master_pod_name -- cat /opt/conjur/etc/ssl/cert/tls.crt)
+
+set_namespace $TEST_APP_NAMESPACE
+
+echo "Storing non-secret Conjur cert as test app configuration data"
+kubectl delete --ignore-not-found=true configmap $TEST_APP_NAMESPACE
+kubectl create configmap $TEST_APP_NAMESPACE --from-file=ssl-certificate=<(echo "$ssl_cert")
+
+echo "Conjur certificate stored."
+
+# Build and push test containers
+announce "Building and pushing test app images"
+
+echo "Building test app image"
+docker build \
+    --build-arg namespace=$TEST_APP_NAMESPACE \
+    --tag test-app:$CONJUR_NAMESPACE \
+    --file test/python/k8s/Dockerfile .
+
+test_app_image="${DOCKER_REG_URL}/test-app:$CONJUR_NAMESPACE"
+docker tag test-app:$CONJUR_NAMESPACE $test_app_image
+docker push $test_app_image
+
+# Deploy test app
+
+deploy_test_app() {
+    kubectl delete --ignore-not-found \
+        deployment/test-app \
+        service/test-app \
+        serviceaccount/test-app
+
+    sleep 5
+
+    conjur_pod="$(kubectl get pods --selector "app=conjur-oss" --no-headers -n $CONJUR_NAMESPACE | awk '{ print $1 }')"
+    conjur_ip="$(kubectl describe pod -n $CONJUR_NAMESPACE $conjur_pod | grep "IP:" | awk '{ print $2 }' | head -1)"
+
+    sed "s#{{ TEST_APP_DOCKER_IMAGE }}#$test_app_image#g" ./test/python/k8s/kubernetes/test-app.yml |
+        sed "s#{{ AUTHENTICATOR_CLIENT_IMAGE }}#$authenticator_client_image#g" |
+        sed "s#{{ IMAGE_PULL_POLICY }}#$IMAGE_PULL_POLICY#g" |
+        sed "s#{{ CONJUR_ACCOUNT }}#$CONJUR_ACCOUNT#g" |
+        sed "s#{{ CONJUR_IP }}#$conjur_ip#g" |
+        sed "s#{{ CONJUR_AUTHN_API_KEY }}#$CONJUR_ADMIN_API_KEY#g" |
+        sed "s#{{ CONJUR_AUTHN_LOGIN_PREFIX }}#$conjur_auth_login_prefix#g" |
+        sed "s#{{ CONJUR_APPLIANCE_URL }}#$conjur_appliance_url#g" |
+        sed "s#{{ CONJUR_AUTHN_URL }}#$conjur_authenticator_url#g" |
+        sed "s#{{ TEST_APP_NAMESPACE_NAME }}#$TEST_APP_NAMESPACE#g" |
+        sed "s#{{ AUTHENTICATOR_ID }}#$AUTHENTICATOR_ID#g" |
+        sed "s#{{ CONFIG_MAP_NAME }}#$TEST_APP_NAMESPACE#g" |
+        sed "s#{{ SERVICE_TYPE }}#NodePort#g" |
+        kubectl create -f -
+
+    echo "Test app deployed."
+}
+
+announce "Deploying Test App Into $TEST_APP_NAMESPACE Namespace"
+
+set_namespace $TEST_APP_NAMESPACE
+
+urlencoded_authn_id=$(urlencode $AUTHENTICATOR_ID)
+authenticator_client_image="cyberark/conjur-authn-k8s-client"
+conjur_appliance_url="https://conjur-oss.$CONJUR_NAMESPACE.svc.cluster.local"
+conjur_authenticator_url="$conjur_appliance_url/authn-k8s/$urlencoded_authn_id"
+conjur_auth_login_prefix=host/conjur/authn-k8s/$AUTHENTICATOR_ID/apps
+IMAGE_PULL_POLICY='Always'
+
+deploy_test_app
+
+echo "Waiting for test-app pod to be ready"
+
+while [[ "$(kubectl get pods -n $TEST_APP_NAMESPACE -o jsonpath={.items[*].status.containerStatuses[0].ready})" != "true" ]]; do
+    echo "."
+    sleep 2
+done
+
+echo "Pod ready"
+
```

Comment:
A few comments explaining some of the finer details of this script (or splitting some of it into functions) would make it a lot more readable for someone new to the project

Reply:
💯 Split the script up into functions, should be a bit more self-evident now